modules/helper.py

```diff
@@ -58,23 +58,73 @@ def getExtension(mime):
 		return None
 
 	@staticmethod
-	def makeFullframe(filename, displayWidth, displayHeight, zoomOnly=False, autoChoose=False):
-		name, ext = os.path.splitext(filename)
-		filename_temp = "%s-frame%s" % (name, ext)
+	def getMimeType(filename):
+		if not os.path.isfile(filename):
+			return None
+
+		cmd = ["/usr/bin/file", "--mime", filename]
+		try:
+			process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+			out, err = process.communicate()
+			mimetype = out.lstrip(filename).lstrip(":").strip()
+			if "; charset=" in mimetype:
+				mimetype = mimetype.split("; charset=")[0]
+		except subprocess.CalledProcessError as e:
+			logging.debug("unable to determine mimetype of file: %s" % filename)
+			return None
+		return mimetype
+
+	@staticmethod
+	def scaleImage(orgFilename, newFilename, newSize):
+		cmd = [
+			'convert',
+			orgFilename + '[0]',
+			'-scale',
+			'%sx%s' % (newSize["width"], newSize["height"]),
+			'+repage',
+			newFilename
+        ]
+
+		try:
+			subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+		except subprocess.CalledProcessError as e:
+			logging.exception('Unable to reframe the image')
+			logging.error('Output: %s' % repr(e.output))
+			return False
+		return True
+		
+
+	# can also be used to check if image file is corrupted
```

Comment:
This function doesn't make me super happy. It has an odd behavior where it will actually delete files if asked to. It's an odd extra job for a function which should only return the size. Worse, the caller won't know if that happened since it is magic 😄 

It's used predominantly by the various content providers, but for them, it would make more sense to do a `os.unlink` on the file in-question by default to make sure it's not there. Or simply allow the request function to overwrite any existing content.

Reply:
You are right, as always ;)
The function doesn't delete files anymore! I moved that part to _useCachedImage_ of the baseService